### PR TITLE
Fix: default frontend to current owner, allow setting no owner on create

### LINKED ISSUE #3346: [BUG] Owner of new correspondent is set automatically, creates visibility issue
### Description

Currently when creating a new correspondent and purposefully leaving the owner empty, it gets set to the current user anyway when saving. In a multi-user environment this means the new correspondent won't be visible to anyone else by default.

To correct this, one has to edit every new correspondent again after creating it and delete the owner.

For me it feels inconsistent if I do not see an owner in the "Create new correspondent"-dialog, but the item gets saved with an assigned owner anyway. Needing to open the same dialog twice to achieve the desired settings is inconvenient.

### Steps to reproduce

1. Go to the "Correspondents"-list.
2. Click "Create"
3. Give it a name and purposefully leave the owner empty to achive the described result:
"Objects without an owner can be viewed and edited by all users"

Screenshot 2023-05-08 094113

4. Click "Save"
5. "Edit" this correspondent
6.  See that an owner was assigned anyway

Screenshot 2023-05-08 094152

7. (Now I need to delete the owner again to make the correspondent visible to anyone as desired)

### Webserver logs

```bash
There are no logs regarding this event.
```

### Browser logs

_No response_

### Paperless-ngx version

Paperless-ngx 1.14.4

### Host OS

Ubuntu 22.04

### Installation method

Docker - official image

### Browser

Chrome

### Configuration changes

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

Fix is two-fold:

- UI should visibly default to current user for object creation
- API / UI should 'override' default owner if the `set_permissions` object explicitly has owner set to empty

Fixes #3346 

https://github.com/paperless-ngx/paperless-ngx/assets/4887959/9444a4d3-925c-46ea-9948-2ac6f8a2a74d

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please explain)

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

File: src-ui/playwright-report/index.html
Comment (stumpylog) [inline src-ui/playwright-report/index.html:1]: Did this file mean to end up in this branch?

Comment (shamoon) [reply src-ui/playwright-report/index.html:1]: Thanks, something else Im working on =)

File: src/documents/tests/test_api.py [hunk 1]
```diff
@@ -3532,6 +3532,44 @@ def test_object_permissions(self):
...
+        self.client.force_authenticate(user1)
+
+        response = self.client.post(
+            "/api/tags/",
+            json.dumps(
+                {
+                    "name": "test1",
+                    "matching_algorithm": MatchingModel.MATCH_AUTO,
+                    "set_permissions": {
+                        "owner": None,
```
Comment (stumpylog) [inline src/documents/tests/test_api.py:3548]: Does a none owner with other permissions make sense?  Wouldn't a none owner be open to all anyway?

Comment (shamoon) [reply src/documents/tests/test_api.py:3548]: Ah, good point. I was trying to be compact but I'll split it up.

File: src/documents/tests/test_api.py [hunk 2]
```diff
@@ -3532,6 +3532,68 @@ def test_object_permissions(self):
             status.HTTP_404_NOT_FOUND,
         )
 
+    def test_api_set_permissions(self):
```
Comment (stumpylog) [inline src/documents/tests/test_api.py:3535]: I know, slightly nit picky, but I do like the given/when stuff

Comment (shamoon) [reply src/documents/tests/test_api.py:3535]: Sure thing!

**Review by stumpylog — APPROVED** [review]

**Pull request merged**